packaged/lib/bxi/base/log/logger.py

```diff
@@ -214,12 +214,26 @@ def _report(self, ei, level, msg, *args, **kwargs):
 
             if not hasattr(value, 'cause') or value.cause is None:
                 break
+            caused_by_str = __BXIBASE_CAPI__.BXIERR_CAUSED_BY_STR
+            caused_by_str_len = __BXIBASE_CAPI__.BXIERR_CAUSED_BY_STR_LEN
+            __BXIBASE_CAPI__.bxierr_report_add(report_c,
+                                               caused_by_str, caused_by_str_len,
+                                               "".encode('utf-8', 'replace'),
+                                               len("".encode('utf-8', 'replace')) + 1)
+
+            # Workaround for Python2/3 compatibility issue :
+            # Using a bytes object for initializing a C string (char*) seems mandatory in
+            # Python 3.
+            # In Python 2, an str object is sufficient.
+            try:
+                cause_str = bytes(str(value.cause), 'utf-8')
+            except TypeError:
+                cause_str = str(value.cause)
 
-            cause_str = __BXIBASE_CAPI__.BXIERR_CAUSED_BY_STR
```

Comment:
formerly named "cause_str" variable is renamed "caused_by_str".
"cause_str" variable in this commit points to the string representation of the "cause" attribute of value, an exception.